application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -173,30 +178,42 @@ private MessageCreateAction createModMessage(String reportReason, String authorN
             .setColor(AMBIENT_COLOR)
             .build();
         return modMailAuditLog.sendMessageEmbeds(reportedMessageEmbed, reportReasonEmbed)
-            .addActionRow(
-                    DiscordClientAction.General.USER.asLinkButton("Author Profile", authorID));
+            .addActionRow(DiscordClientAction.Channels.GUILD_CHANNEL_MESSAGE.asLinkButton(
+                    "Reported Message", guild.getId(), reportMessage.reportedMessageChannelID,
+                    reportMessage.reportedMessageID));
     }
 
     private void sendModMessage(ModalInteractionEvent event, List<String> args,
             TextChannel modMailAuditLog) {
         String reportedMessage = args.get(0);
-        Instant reportedMessageTimestamp = Instant.parse(args.get(1));
-        String authorName = args.get(2);
-        String authorAvatarURL = args.get(3);
-        String authorID = args.get(4);
+        String reportedMessageID = args.get(1);
+        String reportedMessageChannelID = args.get(2);
+        Instant reportedMessageTimestamp = Instant.parse(args.get(3));
+        String reportedMessageAuthorName = args.get(4);
+        String reportedMessageAuthorAvatarUrl = args.get(5);
 
+        ReportMessage reportMessage = new ReportMessage(reportedMessage, reportedMessageID,
+                reportedMessageChannelID, reportedMessageTimestamp, reportedMessageAuthorName,
+                reportedMessageAuthorAvatarUrl);
+
+        Guild guild = event.getGuild();
         event.deferReply().setEphemeral(true).queue();
 
         InteractionHook hook = event.getHook();
         String reportReason = event.getValue(REPORT_REASON_INPUT_ID).getAsString();
 
-        createModMessage(reportReason, authorName, authorAvatarURL, authorID, reportedMessage,
-                reportedMessageTimestamp, modMailAuditLog).mapToResult()
-                    .map(this::createUserReply)
-                    .flatMap(hook::editOriginal)
-                    .queue();
+        createModMessage(reportReason, reportMessage, guild, modMailAuditLog).mapToResult()
+            .map(this::createUserReply)
+            .flatMap(hook::editOriginal)
+            .queue();
     }
 
+    record ReportMessage(String reportedMessage, String reportedMessageID,
+            String reportedMessageChannelID, Instant reportedMessageTimestamp,
+            String reportedMessageAuthorName, String reportedMessageAuthorAvatarUrl) {
+    }
```

Comment:
* since this class provides the context, i guess its obsolete to have the `reported` prefix everywhere on the variables.
* id actually rename the record to `ReportedMessage`, that makes it clear that it represents a bad message. and then just `String content, String messageId, String channelId, Instant messageTimestamp, authorName, authorAvatarUrl`